pkg/rewrite/rewrite.go

```diff
@@ -237,8 +236,10 @@ func Rewrite(rewriteOptions RewriteOptions) error {
 		chartBaseName := v.Spec.Chart.Name
 		// the helmBase may have a chart name prefix removed - we must find the base name instead of the original chart name
 		for _, helmBase := range helmBases {
-			chartName := strings.Split(helmBase.Path, "/")[len(strings.Split(helmBase.Path, "/"))-1]
-			if strings.HasSuffix(chartBaseName, chartName) {
+			baseDir := filepath.Dir(helmBase.Path)
+			chartName := filepath.Base(helmBase.Path)
+			// v is a top level chart, so only top level charts directory name should match
```

Comment:
> so only top level charts directory name should match

how is that enforced here? by `baseDir == "charts"`? isn't there a `charts` directory for each subchart as well?

Reply:
yes, and baseDir is the entire path. it won't match anything nested deeper.